mainsite/selling/forms.py

```diff
@@ -35,6 +36,51 @@ def __init__(self, *args, **kwargs):
         #Gets all the available categories for the new item
         self.fields['category'].queryset = Category.objects.all()
 
+    def clean(self):
+        super().clean()
+
+
+        uplPos = self._request.POST.getlist('uplPos').copy()
+        currUpl = self._request.POST.getlist('currUpl').copy()
+        cleanedPics = []
+        validationErrors = []
+        for pic in self._request.FILES.getlist('curr_picture'):
+            # please, god, deliver these unto me in source ordering
+            #  despite not mentioning this feature anywhere in your holy texts, the Django docs,
+            #   which I guess in this metaphor makes StackExchange the equivalent of rabbinic commentaries
+            try:
+                result = self._clean_a_picture(pic)
+                if result is not None:
+                    cleanedPics.append(result)
+            except forms.ValidationError as e:
+                validationErrors.append(e)
+        if len(validationErrors) > 0:
+            raise forms.ValidationError(validationErrors)
+        if len(cleanedPics) + len(currUpl) == 0:
+            raise forms.ValidationError( "At least one picture is needed!", code="empty")
+
+        item = self.instance  # ?
```

Comment:
Might be more clear if you remove this and replace instances of `item` with `self.instance` 🤷. Up to you.

Reply:
✔️